### PR TITLE
[BUGFIX] QgsMapLayerRegistry: Check layers before removed

### PR DESCRIPTION
**Pull request by rldhont**
Probably fixed #15088 Segmentation fault when using layersRemoved SIGNAL

### CODE REVIEW

File: src/core/qgsmaplayerregistry.cpp [hunk 1]
```diff
@@ -131,17 +131,22 @@ void QgsMapLayerRegistry::removeMapLayers( const QList<QgsMapLayer*>& layers )
...
   }
 
   emit layersWillBeRemoved( layerIds );
-  emit layersWillBeRemoved( layers );
+  emit layersWillBeRemoved( layerList );
 
-  Q_FOREACH ( QgsMapLayer* lyr, layers )
+  Q_FOREACH ( QgsMapLayer* lyr, layerList )
   {
     if ( !lyr )
```
Comment (m-kuhn) [inline src/core/qgsmaplayerregistry.cpp]: This check can be removed now (is redundant with check on 139)

File: src/core/qgsmaplayerregistry.cpp [hunk 2]
```diff
@@ -158,7 +163,7 @@ void QgsMapLayerRegistry::removeMapLayers( const QList<QgsMapLayer*>& layers )
     emit layerRemoved( myId );
   }
 
-  emit layersRemoved( layerIds );
+  emit layersRemoved( layerList );
```
Comment (m-kuhn) [inline src/core/qgsmaplayerregistry.cpp]: I'm not sure that's correct, last time I looked the `layersRemoved` signal was sending passing ids (and not pointers to possibly deleted layers).

File: src/core/qgsmaplayerregistry.cpp (hunk 1 above)
Comment (rldhont) [inline src/core/qgsmaplayerregistry.cpp]: I'll update my proposed code

File: src/core/qgsmaplayerregistry.cpp (hunk 2 above)
Comment (rldhont) [inline src/core/qgsmaplayerregistry.cpp]: It's a mistake sorry, I'll update and rebase my branch

**Comment by rldhont** [comment]
@m-kuhn I have updated the code

**Comment by rldhont** [comment]
@m-kuhn can you review it ? Can I merge and backport it ?

File: src/core/qgsmaplayerregistry.cpp [hunk 3]
```diff
@@ -131,21 +131,23 @@ void QgsMapLayerRegistry::removeMapLayers( const QList<QgsMapLayer*>& layers )
...
+    if ( layer && mMapLayers.contains( layer->id() ) )
+    {
       layerIds << layer->id();
+      layerList << layer;
+    }
   }
 
   emit layersWillBeRemoved( layerIds );
-  emit layersWillBeRemoved( layers );
+  emit layersWillBeRemoved( layerList );
```
Comment (m-kuhn) [inline src/core/qgsmaplayerregistry.cpp]: There is still only one signal defined in the header

    void layersWillBeRemoved( const QStringList& theLayerIds );

**Comment by m-kuhn** [comment]
Looks good to me, thanks!

**Pull request merged**